src/redux/session.js

```diff
@@ -1,21 +1,35 @@
 var keyMirror = require('keymirror');
+var defaultsDeep = require('lodash.defaultsdeep');
 
 var api = require('../mixins/api.jsx').api;
 var tokenActions = require('./token.js');
 
 var Types = keyMirror({
     SET_SESSION: null,
-    SET_SESSION_ERROR: null
+    SET_SESSION_ERROR: null,
+    SET_STATUS: null
 });
 
+module.exports.Status = keyMirror({
+    FETCHED: null,
+    NOT_FETCHED: null,
+    FETCHING: null
+});
+
+module.exports.getInitialState = function (){
+    return {status: module.exports.Status.NOT_FETCHED, session:{}};
+};
+
 module.exports.sessionReducer = function (state, action) {
     // Reducer for handling changes to session state
     if (typeof state === 'undefined') {
-        state = {};
+        state = module.exports.getInitialState();
     }
     switch (action.type) {
     case Types.SET_SESSION:
-        return action.session;
+        return defaultsDeep({results: action.session}, state);
```

Comment:
Just noticed this — this should be updated to `{session: action.session}`, right?